File: 프로그래머스/lv2/42584. 주식가격/주식가격.py
def solution(prices):
    answer = []
    for i in range(len(prices) - 1):
        now = prices[i]
        cnt = 1
        for j in range(i+1, len(prices)):
            if now > prices[j] or j == len(prices) - 1:
                answer.append(cnt)
                break
            else:
                cnt += 1
    answer.append(0)
    return answer

File: 프로그래머스/2/42584. 주식가격/주식가격.java
import java.util.*;
class Solution {
    public int[] solution(int[] prices) {
        ArrayList<Integer> arr = new ArrayList<>();
        for(int i = 0; i < prices.length-1; i++){
            int cnt = 1;
            for(int j = i+1; j < prices.length-1; j++){
                if(prices[j] >= prices[i]){
                    cnt++;
                }else{
                    break;
                }
            }
            cnt = cnt == 0 ? 1 : cnt;
            arr.add(cnt);
        }
        
        int[] ans = new int[prices.length];
        for(int i = 0; i < prices.length-1; i++){
            ans[i] = arr.get(i);
        }
        return ans;
    }
}